spec/features/users/new_spec.rb

```diff
@@ -0,0 +1,22 @@
+require 'rails_helper'
+
+RSpec.describe 'Sign Up', type: :feature do
+  describe 'When user visits sign up page' do
+    it "users can sign up using their google acocunt" do
+      OmniAuth.config.test_mode = true
+      OmniAuth.config.mock_auth[:google_oauth2] = OmniAuth::AuthHash.new({
+        provider: 'google',
+        uid: '123456',
+        info: {
+          name: 'John Doe',
+          email: 'test@example.com'
+        }
+      })
+
+      visit register_user_path
+      click_on "Sign Up with Google"
+
+      expect(current_path).to eq(user_path(User.first))
```

Comment:
Is there a way to test the sad path? Or is this something Selena is doing?

Reply:
there is no sad path for google sign up or sign in. Because if you enter wrong email or password, google will show the message on its sign in page, not our page

Reply:
We'll test a sad path for normal login